meilidb-http/src/routes/index.rs

```diff
@@ -12,104 +17,312 @@ use crate::models::token::ACL::*;
 use crate::routes::document::IndexUpdateResponse;
 use crate::Data;
 
+fn generate_uid() -> String {
+    let mut rng = rand::thread_rng();
+    let sample = b"abcdefghijklmnopqrstuvwxyz0123456789";
+    sample
+        .choose_multiple(&mut rng, 8)
+        .map(|c| *c as char)
+        .collect()
+}
+
 pub async fn list_indexes(ctx: Context<Data>) -> SResult<Response> {
     ctx.is_allowed(IndexesRead)?;
-    let list = ctx
-        .state()
-        .db
-        .indexes_names()
-        .map_err(ResponseError::internal)?;
-    Ok(tide::response::json(list))
+
+    let indexes_uids = ctx.state().db.indexes_uids();
+
+    let env = &ctx.state().db.env;
+    let mut reader = env.read_txn().map_err(ResponseError::internal)?;
+
+    let mut response_body = Vec::new();
+
+    for index_uid in indexes_uids {
+        let index = ctx
+            .state()
+            .db
+            .open_index(&index_uid)
+            .ok_or(ResponseError::internal(&index_uid))?;
```

Comment:
Could you please create a real internal error message please?

Reply:
And use an `ok_or_else` to make sure that the `format` you will call will not be called each time even when no error happen.

Reply:
In this case I think it's probably better to check with `match` if the index exist and continue if not but sent an error log

Reply:
Yes, I aggree.